code_benches/run.py

```diff
@@ -61,6 +71,16 @@ def __init__(self, name, parameters, stack_size, custom_arguments=None, do_lto=T
     def __str__(self):
         return "{}({})".format(self.name, " ".join(map(str, self.parameters)))
 
+    def sources(self):
+        # glob here, avoids issues with glob expansion in shell
+        if self.is_cpp:
+            patterns = ["*.c", "*.cpp"]
+        else:
+            patterns = ["*.c"]
+
+        sources = it.chain.from_iterable(glob.glob(os.path.join(self.name, pattern)) for pattern in patterns)
+        return " ".join(source[len(self.name)+1:] for source in sources)
```

Comment:
just import `chain` directly

also, can you split this into more variables, had some trouble parsing this at first glance

Reply:
Can import chain directly 👍 

Do you have a suggestion where to split this?

I can store a temporary like:
```
paths = (os.path.join(self.name, pattern) for pattern in pattern)
sources = it.chain.from_iterable(glob.glob(path) for path in paths))
```

Does that accomplish what you're looking for?

Unfortunately handling the results from the standard python APIs is rather annoying.